### PR TITLE
Auth smoketests

### PR DESCRIPTION
**Pull request by jasmingacic**
This PR depends on #452 

Basic auth is checked against mocked API

### CODE REVIEW

**Comment by jasmingacic** [comment]
The issue with the global enablement of ext_auth has been fixed by adding 
```
      per_filter_config:
        envoy.filters.http.ext_authz:
          disabled: true
```

**Comment by jasmingacic** [comment]
@mbana @kylehodgetts take a look. I think we could merge this into main instead of auth branch.

File: internal/controllers/envoyfleet_resources.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ func (e *EnvoyFleetResources) generateDeployment(ctx context.Context) error {
 		ImagePullPolicy: corev1.PullIfNotPresent,
 		Command:         []string{"/bin/sh", "-c"},
 		Args: []string{
-			"envoy -c /etc/envoy/envoy.yaml --component-log-level filter:trace,ext_authz:trace --service-node $POD_NAME",
+			"envoy -c /etc/envoy/envoy.yaml --service-node $POD_NAME",
```
Comment (kylehodgetts) [inline internal/controllers/envoyfleet_resources.go:165]: How come the other flags were removed?

File: smoketests/basic_auth/basicauth_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+
+	// v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (kylehodgetts) [inline smoketests/basic_auth/basicauth_test.go:15]: Remove if not needed

**Review by kylehodgetts — CHANGES_REQUESTED** [review]
Looks good. Just need to remove commented out code

File: internal/controllers/envoyfleet_resources.go (hunk 1 above)
Comment (jasmingacic) [reply internal/controllers/envoyfleet_resources.go:165]: I think this was something that @mbana  added while he was writing the feature. 
I just removed --component-log-level so we can catch all logs

File: smoketests/basic_auth/basicauth_test.go (hunk 2 above)
Comment (jasmingacic) [reply smoketests/basic_auth/basicauth_test.go:15]: I forgot to turn on teardown which was using v1. 
Good catch

File: internal/controllers/envoyfleet_resources.go (hunk 1 above)
Comment (jasmingacic) [reply internal/controllers/envoyfleet_resources.go:165]: this is what we have in main branch  `https://github.com/kubeshop/kusk-gateway/blob/main/internal/controllers/envoyfleet_resources.go#L165`

**Review by kylehodgetts — APPROVED** [review]

**Pull request closed without merging**